File: backend/chat-service/src/main/java/com/example/chatservice/chat/entity/ReactionEmoji.java
package com.example.chatservice.chat.entity;

import java.util.Arrays;
import java.util.Optional;

/** 반응 이모지 키. 유니코드 대신 키를 저장하고 앱이 그림으로 바꾼다(👍❤️😂😮😢🙏). */
public enum ReactionEmoji {
    LIKE, HEART, LAUGH, WOW, SAD, PRAY;

    public static Optional<ReactionEmoji> of(String raw) {
        if (raw == null) return Optional.empty();
        return Arrays.stream(values()).filter(e -> e.name().equalsIgnoreCase(raw.trim())).findFirst();
    }
}

File: backend/chat-service/src/main/kotlin/com/example/chatservice/chat/entity/ReactionEmoji.kt
package com.example.chatservice.chat.entity

/** 반응 이모지 키. 유니코드 대신 키를 저장하고 앱이 그림으로 바꾼다(👍❤️😂😮😢🙏). */
enum class ReactionEmoji {
    LIKE, HEART, LAUGH, WOW, SAD, PRAY;

    companion object {
        /** 대소문자·공백을 무시하고 찾는다. 모르는 값이면 null. */
        @JvmStatic
        fun of(raw: String?): ReactionEmoji? {
            if (raw == null) return null
            val key = raw.trim()
            return entries.firstOrNull { it.name.equals(key, ignoreCase = true) }
        }
    }
}
